### DISCUSSION TITLE
Should WordPress get an account on a Mastodon instance?

### DISCUSSION THREAD

**Discussion by sereedmedia** [WordPress/Marketing-Team#168] [Ideas]
Andy Fragan started a thread in the Meta slack meeting posing this question: https://wordpress.slack.com/archives/C02QB8GMM/p1674674187265479

IMO this is important to do in alignment with the Open Source ethos. Hopefully we could just automate the posting as is done with other social networks.

Capturing some of the practical/logistical thoughts and questions raised by @eidolonnight in that same Slack thread for discussion: 

- In the case of getting an account somewhere, the first question is “Which instance?"
- Second, “Is adding yet another network to our list truly needed/impactful?
- It has also been proposed that we set up a WordPress Mastadon instance. In that case, even more than the former, there are time (maintenenace, moderation) and hard costs involved.
- [re: selecting Mastadon instances] how do we fairly pick one to support? And of course what happens if/when the moderators of those instances move on or something? 
- What is the value in supporting any of these given the relative size of Mastadon? Unless all we’re doing in copying tweets or using an RSS bot, the work to support another network is usually >0. If we are are just aggregating posts, in order to keep work at 0, then what value does it offer over something like [#announcements](https://wordpress.slack.com/archives/C024MFP4Q) or any RSS aggregator?

**Comment by courtneyr-dev** [comment]
I think there is value in reaching the open source community already there that may not connect with WordPress on any other social platform. The strategy might look different than just blasting announcement-type content. Perhaps connecting more with other open source platforms such as PHP, Mozilla, etc. would be helpful there.

**Comment by eidolonnight** [comment]
Thanks for kicking this off @sereedmedia . I'd add two notes based on your intro:

> IMO this is important to do in alignment with the Open Source ethos.

I'd like to hear more regarding this.

> Hopefully we could just automate the posting as is done with other social networks.

This process, while aided by Sprout, is far from completely automated. Content is edited, often per network, and then carefully scheduled. Mastodon _could_ be completely automated via RSS bot or something like it, but that is quite different than what we do with the other networks.

**Comment by timiwahalahti** [comment]
There are already plenty of OSS and WordPress contributors, community members and users discussing on Mastodon instead of Twitter/X. It’s a bit silly to not be on a platform that has a big reach in the communities we want to (open source people, developers, individuals whose thoughts align with WordPress's mission to democratize the internet) get involved.

What comes to the selection of instance, I agree that it should be made with care and the project shouldn't probably start running it own due to sysops/developer resources. 

But I think there are already some viable options:
* Fosstodon instance has been running steadily for many years. Other OSS projects use them as well. Their funding is more or less secured and they keep six months' worth of money in the reserve. They have also committed to give warning six months prior in case, for some reason, they switch off the instance.
* Mozzilla has an instance on private beta.
* PHPC Social has a Foundation behind the instance and funding seems to more or less secured.
* Matrix has instance, Foundation behind and I believe funding secured.

**Reply by eidolonnight** [reply]
> a platform that has a big reach in the communities we want to

@timiwahalahti do you have any data that would help size this opportunity?

**Reply by rollecode** [reply]
> > a platform that has a big reach in the communities we want to
> 
> do you have any data that would help size this opportunity?

"Size" as in referring to what exactly? The audience? According to [FediDB](https://fedidb.org) over 10 million people and growing.

**Reply by eidolonnight** [reply]
> "Size" as in referring to what exactly? The audience?

Correct. How many people are on platform, and what is the average follower count for similar brands? It helps to compare to our other platforms so we can objectively weigh the cost and benefit of spending time and resources on potentially adding something like this.

**Reply by rollecode** [reply]
> > "Size" as in referring to what exactly? The audience?
> 
> Correct. How many people are on platform, and what is the average follower count for similar brands? It helps to compare to our other platforms so we can objectively weigh the cost and benefit of spending time and resources on potentially adding something like this.

Well, the first thing what I'm a bit bummed about here is that why is WordPress.org seeking only a commercial, wide-as-possible-reach type of platform in social media when the other major foot is obviously in the open source world? Mastodon and the Fediverse is full of WordPress and PHP people and influencers, WordPress itself connects to ActivityPub and Threads.net connects to ActivityPub as well. So the reach is not any absolute metric here as it's not just one platform. It's the whole web.

For the second question, just see how much [WordPress is talked about](https://mastodon.social/tags/wordpress) between instances, I've seen numerous folks of missing the official WordPress account there, including the community folks. 

Here are some examples of the brand accounts:
- [Internet Archive, 52K followers](https://mastodon.archive.org/@internetarchive)
- [Have I Been Pwned, 14K followers](https://infosec.exchange/@haveibeenpwned)
- [9to5Mac, 25K followers](https://mastodon.online/@9to5Mac)
- [World Wide Web Consortium, 31K followers](https://w3c.social/@w3c)
- [GIMP, 10K followers](https://floss.social/@GIMP)
- [Electronic Frontier Foundation, 62K followers](https://mastodon.social/@eff)
- [1Password, 9.3K followers](https://1password.social/@1password)
- [Signal, 51K followers](https://mastodon.world/@signalapp)

In my mind the worlds largest open source CMS should be a part of this open social web movement.

**Comment by rollecode** [comment]
(Re-started the discussion [here](https://mementomori.social/@rolle/112825973815119247))

To chime in, something like [this](https://x.com/WordPress/status/1815062957232898177) would be definitely nice to see in the Fediverse. WordPress itself has the [ActivityPub support](https://github.com/Automattic/wordpress-activitypub) so why no information gateway through Mastodon as well? There are bots already, naturally, but all the while when updating X you could (and should) post to Mastodon as well? Or even better, just enable ActivityPub on [WordPress.org](https://wordpress.org/) and post from there?

**Reply by jeherve** [reply]
> WordPress itself has the [ActivityPub support](https://github.com/Automattic/wordpress-activitypub) so why no information gateway through Mastodon as well? 
> [...]
> Or even better, just enable ActivityPub on [WordPress.org](https://wordpress.org/) and post from there?

To discuss this option further, I opened a new issue here:
https://github.com/WordPress/wporg-news-2021/issues/425

**Comment by eidolonnight** [comment]
Just to shed some light on the decision to not actively post to Mastodon right now, here are some of the high-level points that have been discussed in Slack and at WordCamps. This is a simplified summary to help centralize conversation, not end it. Please feel free to debate these points. Just because Mastodon isn't supported right now, doesn't mean it won't be supported in the future.

1. Regardless of automation, supporting a platform typically requires human time and effort to periodically check that everything is working well. Even on the platforms that we support currently, little issues pop up all the time. Hashtags might be treated differently, video limits vary, APIs change, etc. While automation is great, the solution here is not as simple as "let a bot do it" (I.e. someone will need to maintain that bot or be accountable for when it inevitably breaks).
2. The nature of the WordPress.org infrastructure has made adding plugin solutions difficult for a variety of reasons. Even existing installed plugins sometimes have unique issues coping with our unique use case, as we've seen with the social features of Jetpack previously. So as much as it would be nice to drop in a plugin, it's rarely that simple.
3. When we do officially support a platform, that means that we are promising to maintain a certain level of brand consistency and content quality. That is partially why we're using Sprout to schedule and manage content, and even with that tool we sometimes fall short. Adding a platform that is not supported by our current software stack makes things even harder. For what it's worth, we are regularly requesting that Sprout add Mastodon support.
4. Both @bjmcsherry and I have looked at Mastodon a couple times in the last year. Despite all the news coverage, the follower counts and engagement we've seen from competing brand accounts were not that impressive. Within the WordPress space, many of the voices there were also present on other platforms as well. That means there would likely be very little incremental value to adding Mastodon as a supported platform, at least for now.
5. In the greater context of WordPress's presence across all social platforms, there are lower effort and higher impact opportunities than Mastodon. The WordPress YouTube channel, for example, has seen exponential growth in the last year. Given limited time and resources, platforms with greater potential are going to receive more attention.
6. To date, all the requests to support Mastodon, that I'm aware of, are because it would be nice to have. None of the proposals have included a unique or novel use case. None of the proposals have provided data showing that Mastodon would allow WordPress to reach a sizable audience that is unreachable through other channels. None of the proposals have included examples of content or experiences that cannot be provided elsewhere.

Those are the points that come to mind right now. These don't just apply to Mastodon either. Most of these points would apply to the addition of any other social network as well. We actually have `WordPress` reserved on Bluesky, but are not actively posting there. Personally, I really like [Vero](https://vero.co/), but it has all the same issues stated above.

**Reply by rollecode** [reply]
Well, I'm kind of disappointed about this approach for commerciality from a CMS that is open source. I've expressed my disappointment [on my instance](https://mementomori.social/@rolle/112949484046173744) as well.

> None of the proposals have included a unique or novel use case.

You could, for example, post things you post elsewhere, for those open source folks and entrepreneurs who only use the Fediverse / Mastodon? Not that hard to support the biggest open source social media network as a biggest open source content management system.

At least for the very minimum: Since you have threads.net account (which is "nice to have" in my world as well, if I put it to your perspective), [please enable Fediverse sharing](https://help.instagram.com/760878905943039) for [@wordpress@threads.net](https://www.threads.net/@wordpress) so we can follow those updates from Mastodon. That doesn't cover those who'se instance has blocked Threads.net though. But it's a start.

**Reply by p37307** [reply]
Enabling Threads.net Fediverse sharing would at least be a good start for them to see what interaction they would get not just from Mastodon but other Fediverse software platforms

**Reply by eidolonnight** [reply]
> I'm kind of disappointed about this approach for commerciality from a CMS that is open source.

WordPress is a big project, and as you've seen, there is plenty of support for the Fediverse in the CMS. As for how people spend their time promoting the brand across social media, and which channels are officially supported, I think it's only natural to consider the points that I listed above. The fact that Mastodon is open source is a very good argument for supporting it, but it's only a single factor. As much as our entire community loves open source, I still see a lot of iPhones and Macbooks at every WordCamp. 😄

> you have threads.net account (which is "nice to have" in my world as well, if I put it to your perspective)

Correct, though in this case it's the support through Sprout that tipped the scales on this one. Also, the adoption there was [quite a bit higher than Mastodon](https://www.threads.net/@zuck/post/C89oeSORn81?xmt=AQGzya_Cp1pvllguZMsj41Ilk3MnGyQyS_Ayh75eaeRQxw). BTW We only recently started experimenting there.

> [please enable Fediverse sharing](https://help.instagram.com/760878905943039) for [@wordpress@threads.net](https://www.threads.net/@wordpress) so we can follow those updates from Mastodon

This might be an option! Tagging in @bjmcsherry to take a look.

**Reply by MsPseudolus** [reply]
One thing the Marketing Team might want to revisit here is the growth trajectory of Mastodon (and the Fediverse, generally). While the current numbers might not be big enough at this exact moment according to your decision-making framework, you could project out when they will be. According to the current growth rate and acceleration of that growth rate, the "magic" number might be due sooner than you think. There's value in being established on a platform just before the hockey stick shoots up, which a lot of indicators suggest is going to happen and may already be happening.

**Comment by MikeTaylor** [comment]
Mastodon is the social network whose values are best aligned with those of WordPress: open, interoperable, a space for ordinary people to bring their own insights on a level playing field with those of big corporations. It would certainly make sense for WordPress to be present on Mastodon.

**Comment by bjmcsherry** [comment]
> [please enable Fediverse sharing](https://help.instagram.com/760878905943039) for [@wordpress@threads.net](https://www.threads.net/@wordpress) so we can follow those updates from Mastodon
>
>This might be an option! Tagging in @bjmcsherry to take a look.

Thanks, @eidolonnight; I have turned on fediverse sharing with the Threads platform. It says that it could take a bit of time to be fully federated, but it should get going fairly quickly.

**Reply by rollecode** [reply]
Thanks!

**Reply by rollecode** [reply]
@wordpressdotcom@threads.net federates already (I think it was enabled earlier), but @wordpress@threads.net is not seen yet, is it enabled?

**Reply by jeherve** [reply]
@ronilaukkarinen Both are enabled, but `@wordpressdotcom` was enabled very early on when Federation came to Threads, while `@wordpress` was enabled yesterday evening. It's going to take a little while for `@wordpress@threads.net` to be fully federated by Threads too.

**Reply by rollecode** [reply]
@jeherve Right, thanks for confirming!

**Reply by rollecode** [reply]
They say it should be a rather fast process but the user still can't be found.

**Reply by bjmcsherry** [reply]
I got a reminder that the service is early beta and will take a bit of time to populate. In the meantime, we are planning for the actual launch with some content to soft launch on [https://mastodon.world/@WordPress](url)

**Reply by rollecode** [reply]
That's so awesome! Thank you so much, let's start drumming 🥁 So elated right now. 🙌

**Comment by bjmcsherry** [comment]
We are beginning to soft launch and test [Mastodon](https://mastodon.world/@WordPress) for WordPress with the idea of testing and experimenting with the platform. Some finetuning is still to happen, but it would be great for everyone to help share and support this new endeavor. :) 

I will close this ticket with this comment. Thanks all!
